webapp/src/app/auth/components/login/login.component.ts

```diff
@@ -43,4 +53,20 @@ export class LoginComponent implements OnInit, OnDestroy {
   get password() {
     return this.loginForm.get('password');
   }
+
+  signInWithGoogle(provider: Provider) {
+    this.store.dispatch(new RedirectWithGoogle('login', provider));
+  }
+
+  afterSignInWithGoogle(event) {
```

Comment:
You can also await for the result from a store dispatch chain if that would make the message passing simpler.

For example: https://github.com/traduora/traduora/blob/master/webapp/src/app/projects/components/add-api-client/add-api-client.component.ts#L67

Reply:
No, because when I change the state I change the one inside the open window and here I use postMessage to send a message to the parent window and then handle the rest